Source/WebCore/Modules/url-pattern/URLPatternCanonical.cpp

```diff
@@ -39,6 +39,16 @@ static bool isInvalidIPv6HostCodePoint(auto codepoint)
     return !isASCIIHexDigit(codepoint) && std::find(validSpecialCodepoints.begin(), validSpecialCodepoints.end(), codepoint) != validSpecialCodepoints.end();
 }
 
+bool isSpecialScheme(StringView input)
```

Comment:
This would definitely need a comment with a link to the spec: https://url.spec.whatwg.org/#special-scheme

But also, it's in the URL spec, not the URLPattern spec, so having the function in url-pattern instead of URL.h seems odd. So I checked the URL code and found: `URL::hasSpecialScheme()`. Let's not duplicate code and reuse it instead.

Reply:
Ah I see. I didn't realize that it was in the URL spec! 

I originally thought that I would just make a small helper method that didn't have an explicit implementation spec in URLPattern. But as you said, I will use `URLParser::isSpecialScheme` instead.